lib/services/fake_app_permissions_service.dart

```diff
@@ -54,19 +48,34 @@ class FakeAppPermissionsService implements AppPermissionsService {
   }
 
   @override
-  Future<bool> isEnabled() async => _enabled;
+  Future<void> dispose() async {
+    await _statusController.close();
+  }
+
+  @override
+  Future<void> init() async {
+    _statusController = StreamController<AppPermissionsServiceStatus>.broadcast(
+      onListen: () async {
+        await Future.delayed(const Duration(milliseconds: 200));
+        _statusController.add(AppPermissionsServiceStatus.enabled(rules));
+      },
+    );
+  }
+
+  @override
+  Stream<AppPermissionsServiceStatus> get status => _statusController.stream;
 
   @override
-  Stream<AppPermissionsServiceStatus> enable() async* {
-    yield AppPermissionsServiceStatusEnabling(0.0);
+  Future<void> disable() async {
+    _statusController.add(AppPermissionsServiceStatus.disabling(0.0));
     await Future.delayed(const Duration(seconds: 3));
```

Comment:
Is this delay necessary?

Reply:
In a sense yes - the fake service exists mainly to test the UI, so there's an artificial delay to simulate the loading state with enough time to observe the UI state